### ISSUE TITLE
Undefined array key when decrypting a row which does not include all encrypted fields

### ISSUE THREAD

**Issue by sfilzek** [paragonie/ciphersweet#80]
I just came across an issue where a exception is thrown when I attempt to decrypt a record which does not include all encrypted fields. This can occur when using ciphersweet in Laravel via spatie/laravel-ciphersweet and not selecting all encrypted DB fields in a query.

The issue seems to be in the decryptRow function of ParagonIE\CipherSweet\EncryptedRow

Here's a snippet to show this issue natively without using Laravel:

```
<?php

use ParagonIE\CipherSweet\CipherSweet;
use ParagonIE\CipherSweet\KeyProvider\StringProvider;
use ParagonIE\CipherSweet\EncryptedRow;

// Setup encryption
$row = (new EncryptedRow(new CipherSweet(new StringProvider('4e1c44f87b4cdf21808762970b356891db180a9dd9850e7baf2a79ff3ab8a2fc')), 'test'))
    ->addTextField('ssn')
    ->addTextField('name')
    ->addTextField('email');

// Encrypt a dummy record
$prepared = $row->prepareRowForStorage([
    'name' => 'John Smith',
    'ssn' => '123-45-6789',
    'email' => 'foo@example.com'
]);

// Grab the record part as it would look coming from a DB query
$dbRecord = $prepared[0];

$decrypted = $row->decryptRow($dbRecord);
/*
    Works as expected..
    [
        "name" => "John Smith",
        "ssn" => "123-45-6789",
        "email" => "foo@example.com",
    ]
*/

// Now assume we are performing a DB query which only selects
// the name and email (not the ssn)
$dbRecordNew = [
    "name" => $dbRecord['name'],
    "email" => $dbRecord['email'],
];

$decrypted = $row->decryptRow($dbRecordNew);

/*
    WARNING  Undefined array key "ssn" in vendor\paragonie\ciphersweet\src\EncryptedRow.php on line 407.

    [
        "name" => "John Smith",
        "email" => "foo@example.com",
        "ssn" => null,
    ]
*/
```

The issue is from line 407 in the decryptRow function of [ParagonIE\CipherSweet\EncryptedRow](https://github.com/paragonie/ciphersweet/blob/53afa7f5e6583b43662d6d1485d4dcca1080839f/src/EncryptedRow.php#L407).

Inserting a check in the decryptRow function to ignoring any non-existent array key would fix this, however I am unsure if this would break anything else.

```
    $key = $this->engine->getFieldSymmetricKey(
        $this->tableName,
        $field
    );
    //---- add this ----------------------
    if (!array_key_exists($field, $row)) {
        continue;
    }
    //------------------------------------
    if (\is_null($row[$field])) {
        $return[$field] = null;
        continue;
    }
```

**Comment by paragonie-security** [comment]
Fixed in v4.2.0

**Comment by thisispiers** [comment]
For my own projects, my solution was to add `requiredWhenDecryptingRow` and `requiredWhenEncryptingRow` flags to `$encryptedRow->addField()`. It adds a little extra configuration for extra flexibility.

See https://github.com/thisispiers/ciphersweet/commit/198e0bf2141db92d376b6b986cdf9392a297eacd
